activestorage/test/models/attached/one_test.rb

```diff
@@ -629,4 +632,14 @@ def avatar
 
     assert_match(/Cannot find variant :unknown for User#avatar_with_variants/, error.message)
   end
+
+  test "using the old signed id for direct uploads attaches the blob" do
+    @was_ignore_blob_id_purpose_for_direct_uploads = ActiveStorage.ignore_blob_id_purpose_for_direct_uploads
+    ActiveStorage.ignore_blob_id_purpose_for_direct_uploads = false
+
+    @user.avatar.attach create_blob(filename: "funky.jpg").signed_id(purpose: :blob_id)
+    assert_equal "funky.jpg", @user.avatar.filename.to_s
+
+    ActiveStorage.ignore_blob_id_purpose_for_direct_uploads = @was_ignore_blob_id_purpose_for_direct_uploads
```

Comment:
Typically the global state reset is ensured with an `ensure`. You can cut down on repeated code by making a helper like this: https://github.com/rails/rails/blob/6ff8725d87dc6f73d0c9401ed9470e7a6dde1409/activestorage/test/test_helper.rb#L103-L110